src/rpc/rawtransaction.cpp

```diff
@@ -811,7 +811,7 @@ UniValue SignTransaction(CMutableTransaction& mtx, const UniValue& prevTxsUnival
                 }
                 Coin newcoin;
                 newcoin.out.scriptPubKey = scriptPubKey;
-                newcoin.out.nValue = 0;
+                newcoin.out.nValue = MAX_MONEY;
```

Comment:
isn't MAX_MONEY *theoretically* a valid amount to send?

Reply:
I left a similar comment in https://github.com/bitcoin/bitcoin/pull/12458#pullrequestreview-135231078, which has been addressed by @Empact in a163673.

Reply:
No, `MAX_MONEY` is defined as 21000000 BTC, but the actual maximum is 20999999.97690000 BTC so you couldn't actually have `MAX_MONEY`.

Reply:
I was wondering about the use of `MAX_MONEY` understanding that it doesn't represent the maximum money supply (`Note that this constant is *not* the total money supply, which in Bitcoin currently happens to be less than 21,000,000 BTC`), but still felt the need to comment on it, because it seemed a valid transaction amount to me on the basis of at least its documentation (`No amount larger than this (in satoshi) is valid.`), the function `MoneyRange` (which defines a range that includes `MAX_MONEY`) and unit test vectors below.

https://github.com/bitcoin/bitcoin/blob/b641f60425674d737d77abd8c49929d953ea4154/src/amount.h#L17-L27

https://github.com/bitcoin/bitcoin/blob/b641f60425674d737d77abd8c49929d953ea4154/src/test/data/tx_valid.json#L78-L80

https://github.com/bitcoin/bitcoin/blob/b641f60425674d737d77abd8c49929d953ea4154/src/test/data/tx_invalid.json#L45-L47

To me the amount `MAX_MONEY` is as valid as the value `0` that it replaces, but admittedly I don't fully comprehend all the details.

Reply:
Please ignore the explanation why the use of `MAX_MONEY` is confusing to me. I needed @sipa's confirmation https://github.com/bitcoin/bitcoin/pull/12458#issuecomment-403743474 to connect the dots.

Reply:
@achow101 True, thanks. Ok with me then.